coding/text_file_reader.cpp

```diff
@@ -0,0 +1,43 @@
+#include "text_file_reader.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+#include "base/string_utils.hpp"
+
+#include <fstream>
+#include <sstream>
+
+namespace text_file_reader
+{
+void ReadLineByLine(std::string const & fileName, Callback && fn)
+{
+  auto const & p = GetPlatform();
+  std::ifstream file(p.WritableDir() + fileName);
```

Comment:
I thought about this method but it's marked as protected and I don't know reasons why it was done.